### PR TITLE
Events UI 🖌️

### LINKED ISSUE #64: Events Screen -> UI and Linking
### Is your feature request related to a problem? Please describe.

Creating the UI of the screen and then linking the completed screen to the correct position in the widget tree of the app.

### PR DESCRIPTION
**Pull request by TreavVasu**
### Description
Events UI

**Type of Change:** (Delete irrelevant options)
- Code
- Testing
- User Interface

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Screenshot_1613704752

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [ ] I have added one/more required dependencies
- [x] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

### CODE REVIEW

File: lib/screens/events/eventsCard.dart [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import 'package:ecellapp/core/res/colors.dart';
```
Comment (werainkhatri) [inline lib/screens/events/eventsCard.dart]: 🔴 
1. file name - `event_card.dart`
2. name of file and name of class in the file should be in sync. You shouldn't call the file `eventsCard` and the class inside, `EventCard`

File: lib/screens/events/eventsCard.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  final Event event;
+
+  const EventCard({Key key, this.event}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    double ratio = MediaQuery.of(context).size.aspectRatio;
+
+    var eventDateTime = event.date.split("T");
+    return Container(
```
Comment (werainkhatri) [inline lib/screens/events/eventsCard.dart]: need for this container?

File: lib/screens/events/eventsCard.dart [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+          child: Stack(
+            children: [
+              Padding(
+                padding: const EdgeInsets.fromLTRB(0, 20, 0, 20),
+                child: Container(
+                  decoration: BoxDecoration(
+                    color: Colors.white,
+                    borderRadius: BorderRadius.circular(22),
+                  ),
+                  // color: Colors.white,
```
Comment (werainkhatri) [inline lib/screens/events/eventsCard.dart]: ?

File: lib/screens/events/eventsCard.dart [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                    gradient: LinearGradient(
+                      colors: <Color>[
+                        C.eventsButtonGradientLeft,
+                        C.eventsButtonGradientMid,
+                        C.eventsButtonGradientMid,
+                        C.eventsButtonGradientRight,
+                        C.eventsButtonGradientRight,
+                      ],
+                    ),
+                    onPressed: () => {},
```
Comment (werainkhatri) [inline lib/screens/events/eventsCard.dart]: add a TODO

**Review by werainkhatri — CHANGES_REQUESTED** [review]
Try to make it as similar to the SpeakerScreen, widget decision wise, as possible.
1. Register Button Gradient does not match with the design
2. No need for a separate gradient button, as you have used it only once
3. Image frame does not look aligned on my phone.
    
4. Reduce padding b/w cards.
5. Time should be in HH:MM AM/PM
6. revert the splash.dart screen.

**Comment by TreavVasu** [comment]
> Try to make it as similar to the SpeakerScreen, widget decision wise, as possible.
> 
>     1. Register Button Gradient does not match with the design
> 
>     2. No need for a separate gradient button, as you have used it only once
> 
>     3. Image frame does not look aligned on my phone.
>        
> 
>     4. Reduce padding b/w cards.
> 
>     5. Time should be in HH:MM AM/PM
> 
>     6. revert the splash.dart screen.

1.  Hope it does now.  
2. Done
3. My device falls in 0.5 <  probably slight error due to your falling for other case please give it a try and upload the SS
4. Done 
5. They are using 24hr notation so keeping just HH: MM
6. Done

**Comment by TreavVasu** [comment]
Screenshot_1613821710

**Comment by werainkhatri** [comment]
1. Bottom left, white card is shown
2. little space b/w image frame and text
3. register button's gradient looks wierd

**Comment by TreavVasu** [comment]
> 
> 
>     1. Bottom left, white card is shown
> 
>     2. little space b/w image frame and text
> 
>     3. register button's gradient looks wierd

On it 3. will take a bit of time will do the update in the next commit

**Comment by TreavVasu** [comment]
> > ```
> > 1. Bottom left, white card is shown
> > 
> > 2. little space b/w image frame and text
> > 
> > 3. register button's gradient looks wierd
> > ```
> 
> On it 3. will take a bit of time will do the update in the next commit

Changes implemented as per the designer. 

Screenshot_1613832839

**Comment by TreavVasu** [comment]
Final Build :

WhatsApp Image 2021-02-20 at 9 59 42 PM

**Comment by TreavVasu** [comment]
I think that it's good enough for the aspect ratio to work but I am not sure for devices <0.5 
- I cannot do anything but guess so I am using heightFactor for fonts as for some device it was giving overflow but it is resolved. 
- It and seems good enough to me (I know with the conversation with you but the ratio isn't good enough I think because for the same device falling into same >0.5 there was still giving the overflow.
- So using hybrid of both i.e. both are used 
- I don't know about positioned element location for ratio<0.5 device (would prefer to change it to heightFactor though)

**Comment by werainkhatri** [comment]
TreavVasu#4

**Review by werainkhatri — APPROVED** [review]

**Pull request merged**